### PR TITLE
CORE-5640: Add GitHub Checks App tools

### PR DESCRIPTION
**Pull request by kathytong**

### CODE REVIEW

**Review by pgerlich — APPROVED** [review]
What will this be used for? If we're going to deliver it to customers, we'll probably want to include a `requirements.txt` if any of these libraries aren't part of the standard python runtime.

**Comment by kathytong** [comment]
> What will this be used for? If we're going to deliver it to customers, we'll probably want to include a `requirements.txt` if any of these libraries aren't part of the standard python runtime.

All part of the Python standard library! Not sure if we need to specify using Python 3 though

File: github-checks-app/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+## Creating a GitHub Enterprise Server App
+1. Run `python github-checks-app-on-enterprise-server.py`
+2. When prompted, input your enterprise server domain
+3. Copy/paste the URL into a browser of your choice, scroll down to click `Create GitHub App`
+4. Take note of the `webhook_secret=<webhook_secret>` in the URL
+    ```
+    > python creating-github-enterprise-server-app.py
+    Enter your GitHub Enterprise Server domain name (i.e. github.companyname.com): github.blucodon.com
+    https://github.blucodon.com...webhook_secret=Z1KPOYifctpzOjfphKj_hqRlZbrDOBG9AU7hgj7iPrk...
```
Comment (paymanblu) [inline github-checks-app/README.md:11]: Is any portion of this secret part of an actual/real secret?

There is no reason to use github.blucodon.com as an example right? it only exposes more of our infrastructure? why not use acme.com?

File: github-checks-app/github-checks-app-on-enterprise-server.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+if __name__ == '__main__':
+    domain = input('Enter your GitHub Enterprise Server domain name (i.e. github.companyname.com): ')
+
+    domain = domain.split('://')[-1]
+    domain = domain.split('/')[0]
+
+    callback_url = f'https://blubracket.blubracket.com/api/github_app/auth?domain={domain}'
+    webhook_url = f'https://blubracket.blubracket.com/api/github_app/events?domain={domain}'
```
Comment (paymanblu) [inline github-checks-app/github-checks-app-on-enterprise-server.py:26]: now that these paths are visible we should ensure that they are rate-limited? @malikoth

**Review by paymanblu — CHANGES_REQUESTED** [review]

**Comment by malikoth** [comment]
> All part of the Python standard library! Not sure if we need to specify using Python 3 though

Python 2 is now officially unsupported by the entire Python community.  I see no reason for us to support it.  It's very much time for everyone to adopt Python 3.

File: github-checks-app/github-checks-app-on-enterprise-server.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    app_webhook_secret = token_urlsafe(nbytes=32)
+
+    dynamic_query_args = f'webhook_secret={app_webhook_secret}&callback_url={callback_url}&webhook_url={webhook_url}'
+
+    url = 'https://blubracket.com'
+    settings = f'name={blubracket_app_name}&request_oauth_on_install=true&public=true&webhook_active=true&url={url}'
+    events = 'checks=write&events[]=check_run&events[]=check_suite&pull_requests=read&events[]=pull_request'
+    static_query_args = f'{settings}&{events}'
+
+    url = f'https://{domain}/settings/apps/new?{dynamic_query_args}&{static_query_args}'
```
Comment (malikoth) [inline github-checks-app/github-checks-app-on-enterprise-server.py:38]:
```suggestion
    query_params = {
        'name': 'BluBracket Checks App',
        'webhook_secret': token_urlsafe(nbytes=32),
        'callback_url': f'https://blubracket.blubracket.com/api/github_app/auth?domain={domain}',
        'webhook_url': f'https://blubracket.blubracket.com/api/github_app/events?domain={domain}',
        'request_oauth_on_install': True,
        'public': True,
        'webhook_active': True,
        'url': 'https://blubracket.com',
        'checks': 'write',
        'events[]': ['check_run', 'check_suite', 'pull_request'],
        'pull_requests': 'read',
    }
    query_param_string = urlencode(query_params, doseq=True, safe=':/?=[]')
    
    url = f'https://{domain}/settings/apps/new?{query_param_string}'
```

File: github-checks-app/github-checks-app-on-enterprise-server.py (hunk 2 above)
Comment (malikoth) [reply github-checks-app/github-checks-app-on-enterprise-server.py:26]: They are.  :)

**Pull request merged**